### PR TITLE
fix: pkg deps don't ignore the workspace project without set version

### PR DESCRIPTION
**Pull request by fireairforce**
closes: #3933

### CODE REVIEW

File: .changeset/dirty-cups-live.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"pkgs-graph": patch
+---
+
+fix: when set workspace protol the pkgs in workspace without version not ignore
```
Comment (07akioni) [inline .changeset/dirty-cups-live.md]:
```suggestion
fix: when set workspace protocol the pkgs in workspace without version not ignore
```

File: packages/pkgs-graph/src/index.ts [hunk 2]
```diff
@@ -91,6 +92,10 @@ export default function <T> (pkgs: Array<Package & T>, opts?: {
           unmatched.push({ pkgName: depName, range: rawSpec })
           return ''
         }
+        if (isWorkspaceSpec && versions.length === 0) {
```
Comment (fireairforce) [inline packages/pkgs-graph/src/index.ts:94]: If the workspace not set version, the Array `versions` will be empty.

File: packages/pkgs-graph/src/index.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ export default function <T> (pkgs: Array<Package & T>, opts?: {
     graph: { [id: string]: PackageNode<T> }
     unmatched: Array<{ pkgName: string, range: string }>
   } {
+    // 路径到 pkg 的映射
```
Comment (zkochan) [inline packages/pkgs-graph/src/index.ts]: Please use English in the comments.

File: packages/pkgs-graph/test/index.ts [hunk 4]
```diff
@@ -669,3 +669,51 @@ test('* matches prerelease versions', () => {
     },
   })
 })
+
+// fix: https://github.com/pnpm/pnpm/issues/3933
+test('create package graph even the package without version', async () => {
```
Comment (zkochan) [inline packages/pkgs-graph/test/index.ts]:
```suggestion
test('successfully create a package graph even when a workspace package has no version', async () => {
```

File: packages/pkgs-graph/src/index.ts (hunk 3 above)
Comment (fireairforce) [reply packages/pkgs-graph/src/index.ts]: sorry, i forget to delete 😂

File: packages/pkgs-graph/test/index.ts (hunk 4 above)
Comment (fireairforce) [reply packages/pkgs-graph/test/index.ts]: Updated~

**Review by zkochan — APPROVED** [review]

**Pull request merged**